R/ncdf.R

```diff
@@ -58,31 +66,42 @@
 #' nc = sf::read_sf(system.file("gpkg/nc.gpkg", package = "sf"), "nc.gpkg")
 #' plot(st_geometry(nc), add = TRUE, reset = FALSE, col = NA)
 read_ncdf = function(.x, ..., var = NULL, ncsub = NULL, curvilinear = character(0),
-    eps = 1e-12, ignore_bounds = FALSE, make_time = TRUE, make_units = TRUE) {
+    eps = 1e-12, ignore_bounds = FALSE, make_time = TRUE, make_units = TRUE,
+    proxy = NULL, downsample = 0) {
 
   if (!requireNamespace("ncmeta", quietly = TRUE))
     stop("package ncmeta required, please install it first") # nocov
   if (!requireNamespace("RNetCDF", quietly = TRUE))
     stop("package RNetCDF required, please install it first") # nocov
 
+  if(inherits(.x, "nc_proxy")) {
+   # maybe do this as a method?
+    nc_request <- get_nc_request(.x)
+    x <- as.character(.x[[1]])
```

Comment:
get_nc_request extracts the nc_request object of each of the stars attributes into a stand alone list.